packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,306 @@
+    <nimble-table-column-mapping field-name="required">
+        Required?
+        <nimble-mapping-boolean key="true">
+            <template>
+                <nimble-icon-check title="Required"></nimble-icon-check>
+            </template>
+        </nimble-mapping-boolean>
+        <nimble-mapping-boolean key="false">
+            <template>
+            </template>
+        </nimble-mapping-boolean>
+    </nimble-table-column-mapping>
+</nimble-table>
+```
+
+Note that the `key` attribute values are always given as strings. In the case of boolean or number mappings, this value is parsed to the appropriate type, so that it can properly be compared to the values from the row records.
+
+```TS
+import {
+    attr,
+    booleanConverter,
+    nullableNumberConverter
+} from '@microsoft/fast-element';
+
+export class TableColumnNumberMapping extends TableColumnMapping {
+    @attr({ converter: nullableNumberConverter })
+    public override key: number | null = null;
+}
+
+export class TableColumnBooleanMapping extends TableColumnMapping {
+    @attr({ converter: booleanConverter })
+    public override key: boolean | null = null;
+}
+
+```
+
+**Alternatives:**
+
+1. This mapping API was originally conceived to implement an icon column type. We could stick to the original, limited scope and define just a `nimble-table-column-icon` (mapping strings to icons). However we will likely need to take a very similar approach to implement columns supporting boolean and enum input types. It would be natural and efficient to have a single column type to support all three, and potentially other use cases in the future.
+
+2. We could make the mapping table column an abstract base type, then derive more specific column types from it, e.g. `nimble-table-column-icon`, `nimble-table-column-enum`, `nimble-table-column-boolean`. However, this doesn't actually make much sense, because there is overlap between the icon column and the enum/boolean columns.
+
+    Mappings of column types:
+
+    - icon column:
+        - number -> icon/spinner
+        - string -> icon/spinner
+        - boolean -> icon/spinner
+    - enum column:
+        - number -> text
+        - string -> text
+        - number -> icon/spinner
+        - string -> icon/spinner
+    - boolean column:
+        - boolean -> text
+        - boolean -> icon/spinner
+
+    An icon column would cover half the use cases of enum and boolean columns. The remaining use cases would be covered by something like a mapped text column. This could be confusing to distinguish from our text column. Additionally, if we are going to have a column that maps numbers/strings/booleans to text, it might as well generalize to icons as well. In which case, we don't need a distinct icon column.
+
+### API
+
+#### Column element:
+
+_Component Name_
+
+-   `nimble-table-column-mapping`
+
+_Props/Attrs_
+
+-   `field-name`: string
+
+_Content_
+
+-   column title (text)
+-   1 or more `nimble-mapping-*` elements
```

Comment:
Do you think these elements would ever be used in a component other than the `nimble-table` or within a column other than the `nimble-table-column-mapping`? If we don't think they'd be used outside of that context, perhaps the name should reflect that. My first thought is `nimble-table-column-mapping-value`, but I could also see the prefix "nimble-table-column" being misleading since it isn't a table column.